pkg/basetypes.go

```diff
@@ -34,3 +34,7 @@ type StaleReportLog struct {
 	Confirmations   int64
 	TransactionHash string
 }
+
+type CheckResult interface {
```

Comment:
interface or struct as in eligible_test.go L13?

Reply:
You can do a struct here with a property of `Eligible`. The reason for the property instead of a function that returns eligibility is to ease marshaling in the event we need to add these values to an observation or outcome. That was the purpose for defining public structs in the refactor with distinct types. Many of these values will need to be unmarshalled in telemetry.